File: src_legacy/org/frankversnel/nl/poortjes/rendering/Renderer.java
package org.frankversnel.nl.poortjes.rendering;

import org.frankversnel.nl.poortjes.component.Transform;
import org.frankversnel.nl.poortjes.rendering.component.Color;

public interface Renderer {

	public void drawRectangle(Transform transform, Color color);

	public void drawCircle(Transform transform, Color color);

	public void drawShape(Transform transform, String shapeId);

	public void clearScreen();

}

File: src/main/scala/org/frankversnel/poortjes/rendering/renderer.scala
package org.frankversnel.poortjes.rendering

import org.frankversnel.poortjes._
import org.frankversnel.poortjes.resource_loading.ResourceId

trait Renderer {
	def drawText(transform: Transform, text: String, color: Color)
	def drawRectangle(transform: Transform, color: Color)
	def drawCircle(transform: Transform, color: Color)
	def drawShape(transform: Transform, resourceId: ResourceId)

	def pushTransform(transform: Transform)
	def popTransform

	def clearScreen
}
